### PR TITLE
Improve regexp_like

### PR DESCRIPTION
**Pull request by zhenxiao**
Fix https://github.com/facebook/presto/issues/2418

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public static Pattern castToRegexp(@SqlType(StandardTypes.VARCHAR) Slice pattern
     @SqlType(StandardTypes.BOOLEAN)
     public static boolean regexpLike(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(RegexpType.NAME) Pattern pattern)
     {
-        return pattern.matcher(source.toString(UTF_8)).find();
+        return Pattern.matches(".*" + pattern.toString() + ".*", source.toString(UTF_8));
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java]: This is much worse.  It recompiles the regular expression for every value.

**Comment by zhenxiao** [comment]
@dain thank you, updated PR

**Comment by dain** [comment]
The problem needs to be fixed in a completely different way.  The problem is we must not compile the pattern within the function.  Instead it needs to be compiled ahead of time in the engine.  We're working on a couple of possible solutions to this problem.  One involves adding a new type, and the other involves switching to Joni.

**Comment by zhenxiao** [comment]
@dain get it. my understanding was incorrect. Let me close this. thank you

**Pull request closed without merging**